run-tests.pl

```diff
@@ -322,19 +325,35 @@ sub preparer
          push @req_futures, $f_start->then( sub {
             my ( $env ) = @_;
 
-            exists $env->{$req} or die "TODO: Missing preparer dependency $req\n";
+            exists $env->{$req} or die "TODO: Missing fixture dependency $req\n";
             Future->done( $env->{$req} );
          });
       }
    }
 
-   return Preparer(
+   return Fixture(
       \@requires,
 
       sub { $f_start->done( @_ ) unless $f_start->is_ready },
 
       Future->needs_all( @req_futures )
-         ->then( $do )
+         ->then( $setup ),
+
+      $teardown ? sub {
+         my ( $self ) = @_;
+         my $result_f = $self->result;
+         $self->result = Future->fail(
+            "This Fixture has been torn down and cannot be used again"
```

Comment:
It fails looking like:

    Running tests/10apidoc/10profile-displayname.pl...
      Testing if: PUT /profile/:user_id/displayname sets my name... PASS
      Testing if: GET /profile/:user_id/displayname publicly accessible... ABORT tests/10apidoc/10profile-displayname.pl because This Fixture has been torn down and cannot be used again at ./run-tests.pl line 350.

Resulting in eventually:

    77 tests FAILED